akka-stream-tests/src/test/scala/akka/stream/ActorMaterializerSpec.scala

```diff
@@ -5,23 +5,83 @@
 package akka.stream
 
 import akka.Done
+import akka.actor.ExtendedActorSystem
+import akka.actor.Extension
+import akka.actor.ExtensionId
+import akka.actor.ExtensionIdProvider
 import akka.actor.{ Actor, ActorSystem, PoisonPill, Props }
 import akka.stream.ActorMaterializerSpec.ActorWithMaterializer
 import akka.stream.impl.{ PhasedFusingActorMaterializer, StreamSupervisor }
 import akka.stream.scaladsl.{ Sink, Source }
 import akka.stream.testkit.{ StreamSpec, TestPublisher }
+import akka.testkit.TestKit
 import akka.testkit.{ ImplicitSender, TestProbe }
 import com.github.ghik.silencer.silent
+import com.typesafe.config.ConfigFactory
 
 import scala.concurrent.Await
+import scala.concurrent.Future
 import scala.concurrent.duration._
 import scala.util.{ Failure, Try }
 
+object IndirectMaterializerCreation extends ExtensionId[IndirectMaterializerCreation] with ExtensionIdProvider {
+  def createExtension(system: ExtendedActorSystem): IndirectMaterializerCreation =
+    new IndirectMaterializerCreation(system)
+
+  def lookup(): ExtensionId[IndirectMaterializerCreation] = this
+}
+
+@silent
+class IndirectMaterializerCreation(ex: ExtendedActorSystem) extends Extension {
+  // extension instantiation blocked on materializer (which has Await.result inside)
+  implicit val mat = ActorMaterializer()(ex)
```

Comment:
wouldn't it be more likely to hit the deadlock if there was a sleep(100) before this line, so that all threads waiting for the extensions are running and are blocked before this?

Reply:
I started out like that but noticed I hit the deadlock consistently without any sleep here so removed it.